### PR TITLE
is Balanced

### PR DESCRIPTION
**Pull request by annamikmorozova**

### CODE REVIEW

File: TreesAndGraphs/isBalanced/solution1.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        new Node(25,
+            null,
+            new Node(31)
+        )
+    )
+);
+
+console.log(isBalanced(tree1)); //true
+console.log(isBalanced(tree5)); //false
+console.log(isBalanced(tree3)); //false
```
Comment (truthgoddess) [inline TreesAndGraphs/isBalanced/solution1.js:75]: @annamikmorozova This is new to me, so I had to do a bit of research, and I could definitely be wrong here, but isn't tree3 balanced? 
aww-board (3)

Comment (truthgoddess) [reply TreesAndGraphs/isBalanced/solution1.js:75]: I agree with your other answers. If the tree I drew is correct, then the smaller numbers represent the height of the left and right subtree for each node, as you can see, for none of the nodes is the difference of subtree height greater than 1. Please correct me if I'm wrong! :)

Comment (annamikmorozova) [reply TreesAndGraphs/isBalanced/solution1.js:75]: I think row 3 has to be filled in order for it to be balanced. And row 4 does not have to be complete

**Review by truthgoddess — COMMENTED** [review]
@anna, can you clarify? the definitions I saw where only to look at a nodes children and see their heights, if their heights are not different than 1 for all nodes then it is balanced. Do you understand it differently?

**Comment by annamikmorozova** [comment]
@truthgoddess Think of it level by level. Every level should have 2^level. Only the last level is the one that is allowed not to be full.

**Comment by truthgoddess** [comment]
Are you thinking about perfectly balanced vs. balanced?

https://webdocs.cs.ualberta.ca/~holte/T26/balanced-trees.html

**Comment by annamikmorozova** [comment]
yes! the reason the one you mentioned is not balanced is because it has two levels that are not full

**Comment by truthgoddess** [comment]
But the problem states:

"For the purposes of this question, a balanced tree is defined to be a tree  such that the heights of the two subtrees of any node never differ by more than one."

As I diagrammed, there is no node whose subtrees have a difference of more than one.

Are there any numbers wrong in my diagram? Sorry if I'm missing something here. I don't think the definition she is using is for a 'perfectly' balanced tree. From what I was reading this morning, perfectly balanced trees are not necessary for optimum time complexity, but balanced like she is talking about it is. 

What am I missing? If my picture is wrong, maybe you can post one to help me explain where I am not understanding?

File: TreesAndGraphs/isBalanced/solution1.js (hunk 1 above)
Comment (truthgoddess) [reply TreesAndGraphs/isBalanced/solution1.js:75]: https://www.youtube.com/watch?v=LU4fGD-fgJQ

**Comment by annamikmorozova** [comment]
Now when I think about it again, I think you are right! Sorry!!!! I think I confused complete and balanced trees <oops>. It also says in the book something like a tree is never really balanced, but more like not terribly unbalanced. So I definitely misunderstood the problem. Sorry again!! Good catch!!

**Comment by truthgoddess** [comment]
@annamikmorozova I gotta say, one of the fun challenges of doing this project with you is the challenge of doing a pull request for something I have never learned, and I have to now research and learn in order to review it. So I really appreciate you setting it up, because I'm definitely no expert, but I wouldn't have had such a fun discussion on coding content without this! :)

**Comment by annamikmorozova** [comment]
Will close it since it is not exactly correct

**Pull request closed without merging**